android/app/src/main/java/org/fossasia/openevent/adapters/GlobalSearchAdapter.java

```diff
@@ -0,0 +1,173 @@
+package org.fossasia.openevent.adapters;
+
+import android.content.Context;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import org.fossasia.openevent.R;
+import org.fossasia.openevent.data.Microlocation;
+import org.fossasia.openevent.data.Speaker;
+import org.fossasia.openevent.data.Track;
+import org.fossasia.openevent.viewholders.TrackViewHolder;
+import org.fossasia.openevent.viewholders.LocationViewHolder;
+import org.fossasia.openevent.viewholders.SpeakerViewHolder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import io.reactivex.disposables.CompositeDisposable;
+
+/**
+ * Created by Shreyas on 5/15/2017.
+ */
+public class GlobalSearchAdapter extends BaseRVAdapter<Object,RecyclerView.ViewHolder>{
+
+    private Context context;
+    private CompositeDisposable disposable;
+
+    private List<Object> filteredResultList = new ArrayList<>();
+
+    //ViewType Constants
+    private final int TRACK = 0;
+    private final int SESSION = 1;
```

Comment:
This was kept so as the Search will get updated to add Sessions within the search itself. It needs some database functions to be added. I thought that could be done in a Separate PR.